src/result.jl

```diff
@@ -373,6 +403,36 @@ function TreeSetColoringResult(
                         queue_end += 1
                         queue[queue_end] = neighbor
                     end
+
+                    # Update lower_triangle_offsets and upper_triangle_offsets
+                    i = leaf
+                    j = neighbor
+                    col_i = view(rv, nzrange(S, i))
+                    col_j = view(rv, nzrange(S, j))
+                    index_offsets += 1
+
+                    #! format: off
+                    # S[i,j] is in the lower triangular part of S
+                    if in_triangle(i, j, :L)
+                        # uplo = :L or uplo = :F
+                        # S[i,j] is stored at index_ij = (S.colptr[j+1] - offset_L) in S.nzval
+                        lower_triangle_offsets[index_offsets] = length(col_j) - searchsortedfirst(col_j, i) + 1
+
+                        # uplo = :U or uplo = :F
+                        # S[j,i] is stored at index_ji = (S.colptr[i] + offset_U) in S.nzval
+                        upper_triangle_offsets[index_offsets] = searchsortedfirst(col_i, j)::Int - 1
```

Comment:
```suggestion
                        upper_triangle_offsets[index_offsets] = searchsortedfirst(col_i, j) - 1
```

Reply:
`JET` is not happy without that. I don't understand why.
It seems related to a type piracy of `DataStructures.jl`.

Reply:
Apparently the type of `col_i` is not properly inferred, which is weird cause it works for `col_j`. Let's leave it like that